src/container/srv_target.c

```diff
@@ -1436,17 +1436,18 @@ cont_child_aggregate(struct ds_cont_child *cont)
 	if (rc)
 		return rc;
 
-	interval = (uint64_t)DAOS_AGG_THRESHOLD * NSEC_PER_SEC;
-	D_ASSERT(hlc > (interval * 2));
+	D_ASSERT(hlc > (*interval * 2));
 	/*
 	 * Assume 'current hlc - interval' as the highest stable view (all
 	 * transactions under this epoch is either committed or aborted).
 	 */
-	epoch_max = hlc - interval;
+	epoch_max = hlc - *interval;
```

Comment:
'epoch_max' shouldn't be changed, it should always be 'current hlc - fixed interval'. (it'll be used to guarantee that all transactions all either committed or to be aborted below this epoch)

Reply:
Not sure I follow, epoch_max is still current hlc - fixed interval here?